Zend/zend_compile.c

```diff
@@ -6218,6 +6221,11 @@ static zend_type zend_compile_typename(
 				zend_error_noreturn(E_COMPILE_ERROR,
 					"Duplicate type %s is redundant", ZSTR_VAL(overlap_type_str));
 			}
+			if ( ((ZEND_TYPE_PURE_MASK(type) & MAY_BE_TRUE) && (single_type_mask == MAY_BE_FALSE))
```

Comment:
I think pretty much all of those parens are unnecessary.

Reply:
May be _technically_ unnecessary, but they make it 100% clear what's grouped with what.